feature/preview/src/main/java/com/google/jetpackcamera/feature/preview/PreviewScreen.kt

```diff
@@ -111,6 +111,7 @@ fun PreviewScreen(
             onChangeDynamicRange = viewModel::setDynamicRange,
             onLowLightBoost = viewModel::setLowLightBoost,
             onChangeImageFormat = viewModel::setImageFormat,
+            onToggleWhenDisabled = viewModel::showToastForDisabledHdrToggle,
```

Comment:
Rather than calling back into the ViewModel, we can just create a model class in the UI layer which the ViewModel can populate. Something like:

```
data class CaptureModeToggleUiState (
  val enabled : Boolean,
  val disabledReason: CaptureModeUnsupportedReason?
  val currentMode: ...
   // All other fields defining the state of the button
) {
  enum CaptureModeUnsupportedReason (val testTag: String, val reasonTextResId: Int) {
    // Enumerations for unsupported reasons
  }
}
```

We can then calculate all the fields needed in `PreviewViewModel` and add `CaptureModeToggleUiState` as a field to `PreviewUiState`.